Hazelcast.Net/Hazelcast.Client/Statistics.cs

```diff
@@ -0,0 +1,296 @@
+            try
+            {
+                _client.GetInvocationService().InvokeOnTarget(request, _ownerAddress);
+            }
+            catch (Exception e)
+            {
+                // suppress exception, do not print too many messages
+                if (Logger.IsFinestEnabled())
+                {
+                    Logger.Finest("Could not send stats ", e);
+                }
+            }
+        }
+
+        private void RegisterMetrics()
+        {
+            _gauges.Add("os.committedVirtualMemorySize",
+                () => Process.GetCurrentProcess().VirtualMemorySize64.ToString());
+            _gauges.Add("os.freePhysicalMemorySize", () => EmptyStatValue);
+            _gauges.Add("os.freeSwapSpaceSize", () => EmptyStatValue);
+            _gauges.Add("os.maxFileDescriptorCount", () => EmptyStatValue);
+            _gauges.Add("os.openFileDescriptorCount", () => Process.GetCurrentProcess().HandleCount.ToString());
+            _gauges.Add("os.processCpuTime", () =>
+            {
+                var totalMilliSec = Process.GetCurrentProcess().TotalProcessorTime.TotalMilliseconds;
+                return ((long) totalMilliSec * 1000000).ToString();
+            });
+            _gauges.Add("os.systemLoadAverage", () => EmptyStatValue); //double value
+            _gauges.Add("os.totalPhysicalMemorySize", () => EmptyStatValue);
+            _gauges.Add("os.totalSwapSpaceSize", () => EmptyStatValue);
+
+            _gauges.Add("runtime.availableProcessors", () => Environment.ProcessorCount.ToString());
+            _gauges.Add("runtime.freeMemory", () => EmptyStatValue);
+            _gauges.Add("runtime.maxMemory", () => Process.GetCurrentProcess().MaxWorkingSet.ToInt64().ToString());
+            _gauges.Add("runtime.totalMemory", () => Process.GetCurrentProcess().WorkingSet64.ToString());
+            _gauges.Add("runtime.uptime",
+                () => ((long) (DateTime.Now - Process.GetCurrentProcess().StartTime).TotalMilliseconds).ToString());
+            _gauges.Add("runtime.usedMemory", () => Process.GetCurrentProcess().WorkingSet64.ToString());
+            _gauges.Add("executionService.userExecutorQueueSize", () => EmptyStatValue);
+        }
+
+        private void FillMetrics(StringBuilder stats, ClientConnection ownerConnection)
+        {
+            AddStat(stats, "lastStatisticsCollectionTime", Clock.CurrentTimeMillis());
+            AddStat(stats, "enterprise", "false");
+            AddStat(stats, "clientType", "CSHARP");
+            AddStat(stats, "clientVersion", DllVersion);
+            AddStat(stats, "clusterConnectionTimestamp", ownerConnection.ConnectionStartTime);
+            AddStat(stats, "clientAddress", ownerConnection.GetLocalSocketAddress());
+            AddStat(stats, "clientName", _client.GetName());
+
+            var credentials = _client.GetClientConfig().GetCredentials();
+//            if (!(credentials is UsernamePasswordCredentials))
```

Comment:
why is this commented out?

Reply:
I thinks we should always send principal. I'll delete the commented code